### PR TITLE
WIP: Add a custom pinch to zoom algorithm

### LINKED ISSUE #404: Zooming via pinch to zoom causes graphical glitches
When zooming via mulittouch pinch to zoom gestures, the page "glitches" it zooms correctly but it also shows the fullscreen view every few moments, which makes zooming really hard.
Also zooming in general causes the view to scroll quite often so even if I'm using the handle in the menu I end up on the last page even if I started in the middle or somewhere.

### PR DESCRIPTION
**Pull request by taaem**
Calculate the zoom value by hand and not use gtk_gesture for zooming.
Currently always zooms to center of screen.
This fixes #404, but the algorithm probably only works for my screen and I would be glad if someone could give some input there.

### CODE REVIEW

File: src/gui/inputdevices/TouchInputHandler.h [hunk 1]
```diff
@@ -11,23 +11,31 @@
...
+	double priLastPosX = -1.0;
+	double priLastPosY = -1.0;
+  double secLastPosX = -1.0;
+  double secLastPosY = -1.0;
+  double lastDiff = 0.0;
 
 private:
 	void actionStart(InputEvent* event);
 	void actionMotion(InputEvent* event);
 	void actionEnd(InputEvent* event);
```
Comment (LittleHuba) [inline src/gui/inputdevices/TouchInputHandler.h]: Rename those to `scroll...` to make their purpose clear as we have two different functionalities in this class now.

File: src/gui/inputdevices/TouchInputHandler.cpp [hunk 2]
```diff
@@ -101,8 +117,71 @@ void TouchInputHandler::actionEnd(InputEvent* event)
...
+	if (this->lastDiff == 0.0)
+	{
+		this->lastDiff = diff;
+		return;
+	}
+
+	ZoomControl* zoomControl = this->inputContext->getView()->getControl()->getZoomControl();
+
+	// TODO: Better algorithm for the zoom value
+	zoomControl->zoomSequenceChange(zoomControl->getZoom() + diff/80000, false);
```
Comment (LittleHuba) [inline src/gui/inputdevices/TouchInputHandler.cpp]: This new algortihm should make sure that the same point of the page stays under the finger when zooming. I'm sure you'll find a way to do this!

**Comment by LittleHuba** [comment]
I'll try it out the next few days!

**Comment by LittleHuba** [comment]
Just tested the functionality. The zooming works good! The things I mentioned above need to be solved though. 

- [ ] Zooming in a natural way that the same point of the page stays under the finger
- [ ] Scrolling while zooming

**Comment by taaem** [comment]
@LittleHuba so technically these commits *should* fix your points, but I'm not totally sure how the scrolling works, because even though the center coordinates should be right it doesn't zoom correctly.
I don't know how zoomWidgetPosX and scrollPositionX  are different. Maybe you know better what is going on.

**Pull request merged**